internal/component/pyroscope/java/loop.go

```diff
@@ -183,7 +183,12 @@ func (p *profilingLoop) start() error {
 		"-o", "jfr",
 	)
 	if cfg.CPU {
-		argv = append(argv, "-e", "itimer")
+		if cfg.Wall {
+			argv = append(argv, "-e", "wall", "-t")
+		} else {
+			argv = append(argv, "-e", "itimer")
+		}
```

Comment:
Since there are many other profiling events, I think we should not have `cfg.Wall bool`, but instead `cfg.Event string` which will default to `itimer` but can be set to any of the values supported. Unless there are some values that don't make sense in the context of Alloy and we should not allow them. cc @simonswine

Reply:
Yes, we could have a `cfg.Event` with `itimer`, `cpu` and `wall` as possible values, and `itimer` as the default. That would be similar with the [java agent configuration](https://grafana.com/docs/pyroscope/latest/configure-client/language-sdks/java/#configuration-options).
>  PYROSCOPE_PROFILER_EVENT | Sets the profiler event. With JFR format enabled, this event refers to one of the possible CPU profiling events: itimer, cpu, wall. The default is itimer.

Is there a way to restrict `cfg.Event` to only those values?

Also, could we have `per thread mode` (`-t`) as a separate flag? From the [async profiler docs](https://github.com/async-profiler/async-profiler?tab=readme-ov-file#wall-clock-profiling) it looks like it's recommended enable it when using the `wall` event.
> Wall-clock profiler is most useful in per-thread mode:` -t`.

Reply:
Yes, you can limit the available options for a `string` field by implementing a `Validate` function on arguments. Example here: https://github.com/grafana/alloy/blob/db2036bde950540900ad1ad4e6c03ebf76c17ad1/internal/component/pyroscope/scrape/scrape.go#L217